src/modules/keyboardmanager/ui/KeyDropDownControl.cpp

```diff
@@ -5,29 +5,34 @@
 #include "BufferValidationHelpers.h"
 #include <common\shared_constants.h>
 #include <common\keyboard_layout_impl.h>
+#include <modules\keyboardmanager\common\Helpers.h>
 
 // Initialized to null
 KeyboardManagerState* KeyDropDownControl::keyboardManagerState = nullptr;
 
-// Get keys code list depending if Disable is in dropdown
-std::vector<DWORD> KeyDropDownControl::GetKeyCodeList(bool isShortcut, bool renderDisable)
+// Get selected value of dropdown or -1 if nothing is selected
+DWORD KeyDropDownControl::GetSelectedValue(ComboBox comboBox)
 {
-    auto list = keyboardManagerState->keyboardMap.GetKeyCodeList(isShortcut);
-    if (renderDisable)
-    {
-        list.insert(list.begin(), CommonSharedConstants::VK_DISABLED);
-    }
+    auto dataContext = comboBox.SelectedValue();
+    if (!dataContext)
+        return -1;
 
-    return list;
+    auto value = winrt::unbox_value<hstring>(dataContext);
+    return stoi(to_string(value));
+}
+
+void KeyDropDownControl::SetSelectedValue(std::wstring value)
+{
+    this->dropDown.as<ComboBox>().SelectedValue(winrt::box_value(value));
```

Comment:
[clarification] What happens if for some reason the value isn't present in the ComboBoxItem list?

Reply:
It will look for a user as nothing is selected. If we call `SelectedValue()` on that dropdown it will return a value that is set even if it is not in the list. Can it ever happen? That situation could have happened with -1, but we never store -1 to config.

Reply:
We update the ComboBox itemSource in the DropDownOpened handler whenever the input language is changed, so I was wondering if that could lead to the issue (since earlier we were just setting the index). I don't think there are cases where we would not use the updated value for this method though, so I think it should be fine.